lib/IMAP/MessageMapper.php

```diff
@@ -149,10 +149,10 @@ public function findAll(Horde_Imap_Client_Socket $client,
 				'ids' => new Horde_Imap_Client_Ids($lower . ':' . $upper)
 			]
 		);
-		if (count($fetchResult) === 0 && $upper < $max) {
```

Comment:
the $upper check didn't make any sense because if we request up to the max UID it's impossible to not get at least one UID. The highest UID must exist if the server just told us that it's there.